http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpClient.java

```diff
@@ -38,4 +41,30 @@
      */
     AbortableCallable<SdkHttpFullResponse> prepareRequest(SdkHttpFullRequest request, SdkRequestContext requestContext);
 
+    /**
+     * Interface for creating an {@link SdkHttpClient} with service specific defaults applied.
+     *
+     * <p>Implementations must tolerate concurrent {@link #build()} and {@link #buildWithDefaults(AttributeMap)} calls from
```

Comment:
Do we need it to be thread safe? If not lets relax this documentation as it's more likely they won't be thread safe with the builder pattern vs the immutable factory pattern.

Reply:
We probably want client builder's `build()` methods usable from multiple threads, so it would be needed here as well. If we're willing to give that up, we can relax it here.

Reply:
I think the builders are inherently not thread safe so we should relax it.